### PR TITLE
v1.14 Backports 2023-08-03

### PR DESCRIPTION
**Pull request by sayboras**
* [x] #27029 (@pchaigno)
    * :warning: there are a couple of conflicts 11f779542d02 and efbba5916347
 * [x] #27115 (@bimmlerd)
     * :warning: small conflict with 825b91d17489
 * [ ] #27141 (@yandzee)
 * [ ] #26744 (@rickysumho)
 * [x] #27122 (@qmonnet)
 * [x] #27089 (@lizrice)
 * [ ] #27090 (@darox)
 * [x] #27177 (@giorio94)
 * [x] #27193 (@nbusseneau)
 * [x] #27172 (@joestringer)
 * [x] #27157 (@sayboras)

PRs skipped due to conflicts:

 * #26638 (@julianwiedmann)
    * :warning: seems like there is dependency with https://github.com/cilium/cilium/pull/26852, which is not marked for backport.

Once this PR is merged, you can update the PR labels via:
```upstream-prs
for pr in 27029 27115 27141 26744 27122 27089 27090 27177 27193 27172 27157; do contrib/backporting/set-labels.py $pr done 1.14; done
```
or with
```
make add-labels BRANCH=v1.14 ISSUES=27029,27115,27141,26744,27122,27089,27090,27177,27193,27172,27157
```

### CODE REVIEW

**Review by qmonnet — APPROVED** [review]
> #27122

looks good, thank you!

**Review by bimmlerd — APPROVED** [review]
#27115 lgtm

File: daemon/cmd/status.go [hunk 1]
```diff
@@ -521,6 +521,36 @@ func (c *clusterNodesClient) NodeConfigurationChanged(config datapath.LocalNodeC
...
+
+func (c *clusterNodesClient) DumpNodeIDs() []*models.NodeID {
+	// no-op
+	return nil
+}
+
+func (c *clusterNodesClient) RestoreNodeIDs() {
+	// no-op
+	return
+}
```
Comment (pchaigno) [inline daemon/cmd/status.go]: Commit `datapath, ipcache: Remove unused node ID functions`, I believe the 6 functions above were backported by mistake.

**Review by pchaigno — COMMENTED** [review]
Thanks for backporting my behemoth of a PR!

I believe I found one issue below. Also, were there conflicts in any other commits?

File: daemon/cmd/status.go (hunk 1 above)
Comment (sayboras) [reply daemon/cmd/status.go]: good catch :+1:

Comment (sayboras) [reply daemon/cmd/status.go]: there are a couple of commits having conflicts, however, this is the first time I am using backport tool, so didn't note down :(

**Comment by sayboras** [comment]
/test-backport-1.14

File: daemon/cmd/status.go (hunk 1 above)
Comment (sayboras) [reply daemon/cmd/status.go]: Managed to find the conflict commit in my terminal, I have updated the PR description with more details.

**Review by rolinh — APPROVED** [review]
API changes lgtm.

File: daemon/cmd/status.go (hunk 1 above)
Comment (pchaigno) [reply daemon/cmd/status.go]: Thanks Tam!

Yep, I don't use the backporter tool for exactly that reason. I like the text file of the old backporting scripts for note taking.

File: bpf/lib/common.h [hunk 2]
```diff
@@ -610,7 +610,9 @@ enum {
 #define DROP_INVALID_CLUSTER_ID	-192
 #define DROP_DSR_ENCAP_UNSUPP_PROTO	-193
 #define DROP_NO_EGRESS_GATEWAY	-194
+#define DROP_UNENCRYPTED_TRAFFIC -195
```
Comment (pchaigno) [inline bpf/lib/common.h]: I believe this one was added by mistake. Probably wouldn't have happened if I didn't decide to fix an indentation in the upstream commit :facepalm:

Comment (sayboras) [reply bpf/lib/common.h]: ah good catch, thanks :man_facepalming:

**Comment by sayboras** [comment]
/test-backport-1.14

**Review by pchaigno — APPROVED** [review]

**Review by nbusseneau — APPROVED** [review]
LGTM.

**Comment by sayboras** [comment]
/test-backport-1.14

**Comment by sayboras** [comment]
/test-backport-1.14

**Comment by sayboras** [comment]
/test-backport-1.14

**Review by giorio94 — APPROVED** [review]
My commit looks good. Thanks!

**Review by lizrice — APPROVED** [review]
#27089 looks good

**Review by joestringer — APPROVED** [review]

**Comment by sayboras** [comment]
/test-backport-1.14

**Comment by sayboras** [comment]
Due to a huge number of commits in this backport, build-commit GHA timed out after 1.5h. Hence, I manually verify with the below commands:

```bash
$ time git rebase --exec "make build -j $(nproc)" e391c76420
...
Successfully rebased and updated refs/heads/pr/v1.14-backport-2023-08-03.
git rebase --exec "make build -j $(nproc)" e391c76420  2836.36s user 532.20s system 1471% cpu 3:48.88 total

$ time git rebase --exec "make -C bpf build_all -j $(nproc)" e391c76420
...
Successfully rebased and updated refs/heads/pr/v1.14-backport-2023-08-03.
git rebase --exec "make -C bpf build_all -j $(nproc)" e391c76420  6199.31s user 190.38s system 278% cpu 38:17.06 total

```

**Comment by sayboras** [comment]
Most of the reviews are in, CI is green, marking this ready to merge.

**Pull request merged**